### PR TITLE
FindImageFromTagsTask fails task if no tagged images are found.

### PR DESCRIPTION
**Pull request by zanthrash**
Presently we SUCCEED the task/stage and the downstream stage will bomb out.

@spinnaker/netflix-reviewers PTAL

### CODE REVIEW

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/image/FindImageFromTagsTask.java [hunk 1]
```diff
@@ -47,11 +48,27 @@ public TaskResult execute(Stage stage) {
 
     StageData stageData = (StageData) stage.mapTo(StageData.class);
     Collection<ImageFinder.ImageDetails> imageDetails = imageFinder.byTags(stage, stageData.packageName, stageData.tags);
+
+    if(imageDetails == null || imageDetails.isEmpty()) {
```
Comment (ajordens) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/image/FindImageFromTagsTask.java]: Why not just throw an exception here?

Comment (ajordens) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/image/FindImageFromTagsTask.java]: (minor) space between `if` and `(`

**Comment by zanthrash** [comment]
@ajordens exception thrown

File: orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/image/FindImageFromTagTaskSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import com.netflix.spinnaker.orca.ExecutionStatus
+import com.netflix.spinnaker.orca.clouddriver.tasks.providers.aws.AmazonImageFinder
+import com.netflix.spinnaker.orca.pipeline.model.AbstractStage
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import spock.lang.Specification
+
+
+class FindImageFromTagTaskSpec extends Specification {
+
+  static class TestStage extends AbstractStage {}
```
Comment (ajordens) [inline orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/image/FindImageFromTagTaskSpec.groovy]: Should be able to just use a `PipelineStage` ...



```

def imageFinder = Mock(ImageFinder)

@Subject
def task = new FindImageFromTagsTask(imageFinders: [imageFinder])

def "..."() {
   given:
   def pipelineStage = new PipelineStage(new Pipeline(), "", [packageName: 'myPackage', tags: ['foo': 'bar'])

   when:
   def result = task.execute(stage)

   ...
}
```

**Comment by zanthrash** [comment]
@ajordens test un-biggend

**Comment by ajordens** [comment]
nice! LGTM

**Pull request merged**